CRM/Member/Form/MembershipType.php

```diff
@@ -54,6 +54,13 @@ public function preProcess() {
     $url = CRM_Utils_System::url('civicrm/admin/member/membershipType', 'reset=1');
     $session->pushUserContext($url);
 
+    // when custom data is included in this page
```

Comment:
Could we make this part of the parent preProcess? I think the idea is that the form would have entityType on it & we would via the preProcess handle it.

Reply:
@eileenmcnaughton You mean put it right up into `CRM_Core_Form`?  We could do that and if getDefaultEntity does not throw an exception then run the custom data code.  Would need to check that doesn't break any existing forms which might then run the custom code twice.